### ISSUE TITLE
batch_size_rows: 50,000 fails to copy last batch when performing log_initial

### ISSUE THREAD

**Issue by johnoscott** [transferwise/pipelinewise#240]
My initial tap-run of Oracle -> Snowflake is skipping the last batch of the initial load sync method (log_initial). I built a docker instance from master branch  ( Oct 12, 2019 commit cdc92fca094624c864f171b29d96dcafa34173c7 ). The symptom is that the target row counts are a multiple of the batch_size_rows: 50000 (which I increased from the default 20,000) and the last partial batch are missing. For example
TABLE       ORACLE ROW COUNT       SNOWFLAKE ROW COUNT
A                 3,180,921             3,150,000
B                    50,000                59,343
NOTE This is a pre-release version so might be a known defect. 

When I reverted  batch_size_rows to the default of 20,000 the problem went away.

**Comment by johnoscott** [comment]
Here is a log snippet that shows the behaviour ```[ec2-user .pipelinewise]$ tail -f prod-myschema-snowflake/prod-myschema/log/*.running
INFO dsn: (DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=10.0.0.0)(PORT=1521))(CONNECT_DATA=(SID=SID)))
INFO End SCN: 39991613740
INFO No currently_syncing found
INFO Beginning sync of stream(MYSCHEMA-TBLTOPIC) with sync method(log_initial)
INFO stream MYSCHEMA-TBLTOPIC is using log_miner. will use full table for first run
INFO dsn: (DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=10.0.0.0)(PORT=1521))(CONNECT_DATA=(SID=SID)))
INFO select SELECT  "ACTIVE" , "TOPICCODE" , "TOPICEMAIL" , "TOPICID" , "TOPICNAME" ,to_char( "APPOINTEDDATE" ), "BILLINGCODE" , "BRANDDIR" , "BUSINESSNUMBER" ,to_char( "CHANGEDDATE" ), "CHANGEDUID" , "COMMENTS" ,to_char( "CREATEDDATE" ), "CREATEDUID" , "CUSTOMTOPICID" , "CUSTOMGUID" ,to_char( "DATECHANGED" ), "HEADTOPICID" , "ISHEADOFFICE" , "ISMASTERTOPIC" , "ISPROSPECTIVE" , "LEGALNAME" , "LOGO" , "MASTERTOPICID" , "OFFICENAME" , "ORGANISATIONTYPEID" , "PREVENTAPPLICATIONS" , "PRIMARYCONTACTID" , "TOEXPORT" , "WEBSITE" , ORA_ROWSCN
                                FROM MYSCHEMA.TBLTOPIC
                               ORDER BY ORA_ROWSCN ASC
INFO Getting catalog objects from information_schema cache table...
INFO Schema 'PROD_MYSCHEMA' does not exist. Creating... CREATE SCHEMA IF NOT EXISTS PROD_MYSCHEMA
INFO Table 'PROD_MYSCHEMA.tbltopic' exists
INFO Uploading 50000 rows to external snowflake stage on S3
INFO Target S3 bucket: studylink-roundrect-data, local file: /tmp/tmpmxik18ty, S3 key: prod-pipelinewise-snowflake-imports/pipelinewise_MYSCHEMA-TBLTOPIC_20191014-222240-444265.csv
INFO Loading 50000 rows into 'PROD_MYSCHEMA.tbltopic'
INFO SNOWFLAKE - Merge into PROD_MYSCHEMA.tbltopic: [{'number of rows inserted': 50000, 'number of rows updated': 0}]
INFO Deleting prod-pipelinewise-snowflake-imports/pipelinewise_MYSCHEMA-TBLTOPIC_20191014-222240-444265.csv from external snowflake stage on S3
INFO METRIC: {"type": "counter", "metric": "record_count", "value": 59343, "tags": {}}
INFO Beginning sync of stream(MYSCHEMA-TBLSUBJECT) with sync method(log_initial)
INFO stream MYSCHEMA-TBLSUBJECT is using log_miner. will use full table for first run
INFO dsn: (DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=10.0.0.0)(PORT=1521))(CONNECT_DATA=(SID=SID)))
INFO Schema 'PROD_MYSCHEMA' does not exist. Creating... CREATE SCHEMA IF NOT EXISTS PROD_MYSCHEMA
INFO select SELECT  "TOPICAPPLICATIONID" , "TOPICAPPLICATIONSTATUSID" , "TOPICID" , "AGREEDTERMSCONDITIONS" , "SUBJECTCODE" , "SUBJECTID" , "SUBJECTPORTALID" , "SUBJECTSTATUS" , "BRANCHID" , "BRANDINGID" ,to_char( "CHANGEDDATE" ), "CHANGEDUID" , "COUNSELLORID" ,to_char( "CREATEDDATE" ), "CREATEDUID" , "ENQUIRYSOURCEID" , "ISARCHIVEDSUBJECT" , "KEYWORD" , "PROVIDERSUBJECTID" , "REFERRALSITEID" , "USERID" , "WESTERNUNIONUSERNAME" , ORA_ROWSCN
                                FROM MYSCHEMA.TBLSUBJECT
                               ORDER BY ORA_ROWSCN ASC
INFO Table 'PROD_MYSCHEMA.tblsubject' exists```

**Comment by koszti** [comment]
This is a normal behaviour. Not full batches are flushing in parallel at the end of the tap run. This gives better performance when loading multiple tables by sending multiple queries (MERGE command) to snowflake in parallel.
